test/OpenTelemetry.Tests/Metrics/MetricTestsBase.cs

```diff
@@ -23,6 +32,75 @@ public class MetricTestsBase
     public const string EmitOverFlowAttributeConfigKey = "OTEL_DOTNET_EXPERIMENTAL_METRICS_EMIT_OVERFLOW_ATTRIBUTE";
     public const string ReclaimUnusedMetricPointsConfigKey = "OTEL_DOTNET_EXPERIMENTAL_METRICS_RECLAIM_UNUSED_METRIC_POINTS";
 
+    protected readonly IConfiguration configuration;
+
+    protected MetricTestsBase()
+    {
+    }
+
+    protected MetricTestsBase(IConfiguration configuration)
+    {
+        this.configuration = configuration;
+    }
+
+#if BUILDING_HOSTING_TESTS
+    public static IHost BuildHost(
```

Comment:
Could we move these to `OpenTelemetry.Tests\Shared` and have `OpenTelemetry.Tests` add a ProjectReference to `OpenTelemetry.Extensions.Hosting`? Would that help us get rid of the new preprocessor directive and the compile include statements?

Reply:
The preprocessors are there not to get around dependency issues. I wanted to run the tests in two different modes. In OpenTelemetry.Tests they are run using the traditional SDK + `AddMeter` API. In OpenTelemetry.Extensions.Hosting.Tests they are run using the `IMetricsListener` API and `AddMeter` becomes `services.AddMetrics(builder => builder.EnableMetrics(name));`.

After this PR I need to do one for `WithLogging`. After that I want to explore moving `OpenTelemetryBuilder` into SDK. What that will do is give us a nice thing to extend for cross-cutting registrations like #4940. I do think this is a bit messy and could likely be cleaned up but I think maybe wait to refactor it until that is settled?